src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -384,15 +385,28 @@ internal partial class UnboundLambda
             ImmutableArray<TypeWithAnnotations> types,
             ImmutableArray<string> names,
             ImmutableArray<bool> discardsOpt,
+            ImmutableArray<EqualsValueClauseSyntax?> defaultValues,
             bool isAsync,
             bool isStatic)
         {
             Debug.Assert(binder != null);
             Debug.Assert(syntax.IsAnonymousFunction());
             bool hasErrors = !types.IsDefault && types.Any(static t => t.Type?.Kind == SymbolKind.ErrorType);
 
+            // If we have a parenthesized lambda or anonymous method expression,
+            // we could have default parameters and thus we include the parameter list syntax
+            // to re-use existing binder code for type-checking default params
+            var paramSyntaxList = syntax switch
```

Comment:
>var paramSyntaxList = syntax switch [](http://example.com/codeflow?start=12&length=35)

I'm not sure if we should be binding default values here (within UnboundLambda) or in the caller. (We should probably decide that first.) But if it's here and we need the syntax for individual parameters, should the caller provide that directly?

Reply:
The unbound lambda depends on the syntax node already, so it seemed better to just make use of the respective properties from the underlying syntax node

Reply:
It looks like there's only one caller of `UnboundLambda.Create()` and that caller already switches on the anonymous function syntax kind to get other values. I think for consistency, we should handle the parameter syntax there.